src/OnboardingSPA/components/GlobalStyleParent/index.js

```diff
@@ -0,0 +1,74 @@
+import { useSelect, useDispatch } from '@wordpress/data'; 
+import { useState, useEffect } from '@wordpress/element'; 
+
+import { store as nfdOnboardingStore } from '../../store';
+import { getGlobalStyles, setGlobalStyles } from '../../utils/api/themes';
+import { useGlobalStylesOutput } from '../../utils/global-styles/use-global-styles-output';
+
+/**
+ * Global Style Parent Component
+ * The Fetching of Global Style Object from either store or API is 
+ * common to a lot many places and this component does the trick import { useState, useEffect } from '@wordpress/element';for us.
+ *
+ * @returns Global Style Parent
+ */
+
+const GlobalStyleParent = ({ children }) => {
+
+    const [isLoaded, setIsLoaded] = useState(false);
+
+    const {
+        currentData,
+        storedPreviewSettings,
+    } = useSelect((select) => {
+        return {
+            currentData:
+                select(nfdOnboardingStore).getCurrentOnboardingData(),
+            storedPreviewSettings:
+                select(nfdOnboardingStore).getPreviewSettings(),
+        };
+    }, []);
+
+    const { updateThemeStatus, updatePreviewSettings } =
+        useDispatch(nfdOnboardingStore);
+
+    const getStylesAndPatterns = async () => {
+        const globalStyles = await getGlobalStyles();
+        if (globalStyles?.error) {
+            return updateThemeStatus(THEME_STATUS_NOT_ACTIVE);
+        }
+        let selectedGlobalStyle;
+        if (storedPreviewSettings?.title)
+            selectedGlobalStyle = storedPreviewSettings;
+        else if (currentData.data.theme.variation) {
+            selectedGlobalStyle = globalStyles.body.filter(
+                (globalStyle) =>
+                    globalStyle.title === currentData.data.theme.variation
+            )[0];
+        } else {
+            selectedGlobalStyle = globalStyles.body[0];
+        }
+        
+        if(selectedGlobalStyle)
+            setGlobalStyles({
+                ...selectedGlobalStyle,
+                'title': currentData.data.theme.variation,
+                'version': 2
+            });
+
+        updatePreviewSettings(
+            useGlobalStylesOutput(selectedGlobalStyle, storedPreviewSettings)
+        );
+        setIsLoaded(true);
```

Comment:
Is there a reason to get all variations and then filter out the selected one, if I am not wrong a single API call will get us the current `settings` object from the option?

Reply:
So basically that won't happen unless we are on colors page, till then this will act like before and throw 6 variations, but later just one so iterating later also wont be a task as the first one will match.